internal/guest/storage/overlay/overlay.go

```diff
@@ -22,6 +27,36 @@ var (
 	unixMount   = unix.Mount
 )
 
+// processErrNoSpace logs disk space and inode information for `path` that we encountered the ENOSPC error on. This can be used to get a better
+// view of whats going on on the disk at the time of the error.
+func processErrNoSpace(ctx context.Context, path string, err error) {
+	st := &unix.Statfs_t{}
+	// Pass in filepath.Dir() of the path as if we got an error while creating the directory it definitely doesn't exist. Take its parent
+	// which should be on the same drive.
+	if statErr := unix.Statfs(filepath.Dir(path), st); statErr != nil {
+		log.G(ctx).WithError(statErr).WithField("path", filepath.Dir(path)).Warn("failed to get disk information for ENOSPC error")
+		return
+	}
+
+	all := st.Blocks * uint64(st.Bsize)
+	available := st.Bavail * uint64(st.Bsize)
+	free := st.Bfree * uint64(st.Bsize)
+	used := all - free
+
+	toGigabyteStr := func(val uint64) string {
+		return fmt.Sprintf("%.1f", float64(val)/float64(memory.GigaByte))
+	}
+
+	log.G(ctx).WithFields(logrus.Fields{
+		"available-disk-space-GB": toGigabyteStr(available),
+		"free-disk-space-GB":      toGigabyteStr(free),
+		"used-disk-space-GB":      toGigabyteStr(used),
```

Comment:
To be super pedantic, this should be `GiB`

Reply:
Will change!